backend/accounts/notes/notes.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package notes provides functionality to retrieve and store account transaction notes.
+package notes
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
+)
+
+// NotesData is the notes JSON data serialized to disk.
+type notesData struct {
+	// More fields to be added when we can label more stuff, e.g. receive addresses, utxos, etc.
+
+	// a map of transaction ID to transaction note.
+	TransactionNotes map[string]string `json:"transactions"`
+}
+
+// read deserializes the json files into notes. If the file does not exist yet, no error is
+// returned, and the struct is retruned with default values.
+func read(filename string) (*notesData, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return &notesData{}, nil
+		}
+		return nil, errp.WithStack(err)
+	}
+	defer file.Close() //nolint:errcheck
+	var notes notesData
+	if err := json.NewDecoder(file).Decode(&notes); err != nil {
+		return nil, errp.WithStack(err)
+	}
+	return &notes, nil
+}
+
+func write(data *notesData, filename string) error {
+	file, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+	encoder := json.NewEncoder(file)
+	encoder.SetIndent("", "  ")
+	if err := encoder.Encode(data); err != nil {
+		return errp.WithStack(err)
+	}
+	return nil
+}
+
+// Notes is a high level helper for notes, allowing you to read and set notes for transactions.
+type Notes struct {
+	filename string
+	data     *notesData
+	dataMu   sync.RWMutex
+}
+
+// LoadNotes makes a new Notes instance, already pre-loading all notes into RAM. If the file does
+// not exist, no error is returned.  Returns an error for other kinds of file read errors.
+func LoadNotes(filename string) (*Notes, error) {
+	notes := &Notes{
+		filename: filename,
+	}
+	data, err := read(filename)
+	if err != nil {
+		return nil, err
+	}
+	notes.data = data
```

Comment:
Or you could create the struct right here, when everything's ready:

```go
return &Notes{data: data}, nil
```

Reply:
very good! fixed